plugins/linear-genome-view/src/BaseLinearDisplay/models/BaseLinearDisplayModel.tsx

```diff
@@ -333,8 +334,15 @@ export const BaseLinearDisplay = types
         const featureWidget = session.addWidget(
           'BaseFeatureWidget',
           'baseFeature',
-          { featureData: feature.toJSON(), view: getContainingView(self) },
+          {
+            view: getContainingView(self),
+            display: self,
+            track: getContainingTrack(self),
+          },
         )
+
+        // @ts-ignore
+        featureWidget.setFeatureData(feature.toJSON())
```

Comment:
instead of passing featureData in, I use the action here. this is notable because calling the action is needed to call the customized callbacks. alternatively could use e.g. an afterCreate action and mutate it after loading with the customized callbacks.

Reply:
I now use an autorun to allow featureData passed into model.create to get formatted. this uses a couple tricks

1) preProcessSnapshot converts featureData passed into model.create into "unformattedFeatureData"
2) then autorun processes "unformattedFeatureData" into "featureData"

this allows us to continue to refer to featureWidget.featureData as usual, and not have to call special actions to update the feature data